File: Exporter/Backend/src/modules/transport_layer/domain_value/character_gear.rs
use crate::modules::transport_layer::CharacterItemDto;

#[derive(Debug, Clone, Serialize)]
pub struct CharacterGearDto {
    pub head: Option<CharacterItemDto>,
    pub neck: Option<CharacterItemDto>,
    pub shoulder: Option<CharacterItemDto>,
    pub back: Option<CharacterItemDto>,
    pub chest: Option<CharacterItemDto>,
    pub shirt: Option<CharacterItemDto>,
    pub tabard: Option<CharacterItemDto>,
    pub wrist: Option<CharacterItemDto>,
    pub main_hand: Option<CharacterItemDto>,
    pub off_hand: Option<CharacterItemDto>,
    pub ternary_hand: Option<CharacterItemDto>,
    pub glove: Option<CharacterItemDto>,
    pub belt: Option<CharacterItemDto>,
    pub leg: Option<CharacterItemDto>,
    pub boot: Option<CharacterItemDto>,
    pub ring1: Option<CharacterItemDto>,
    pub ring2: Option<CharacterItemDto>,
    pub trinket1: Option<CharacterItemDto>,
    pub trinket2: Option<CharacterItemDto>,
}

File: Webclient/src/app/module/armory/domain_value/character_gear.ts
import {CharacterItem} from "./character_item";

export interface CharacterGear {
    id: number;
    head: CharacterItem | null;
    neck: CharacterItem | null;
    shoulder: CharacterItem | null;
    back: CharacterItem | null;
    chest: CharacterItem | null;
    shirt: CharacterItem | null;
    tabard: CharacterItem | null;
    wrist: CharacterItem | null;
    main_hand: CharacterItem | null;
    off_hand: CharacterItem | null;
    ternary_hand: CharacterItem | null;
    glove: CharacterItem | null;
    belt: CharacterItem | null;
    leg: CharacterItem | null;
    boot: CharacterItem | null;
    ring1: CharacterItem | null;
    ring2: CharacterItem | null;
    trinket1: CharacterItem | null;
    trinket2: CharacterItem | null;
}
